### PR TITLE
Feat/#120 API endpoint for retrieving the profile data of the currently authenticated user

### PR DESCRIPTION
**Pull request by SamixYasuke**
# Implements API endpoint for retrieving the profile data of the currently authenticated user

## Changes proposed
- Added view user profile endpoint

## Check List (Check all the applicable boxes)

:rotating_light:Please review the [style guide for contributing](add link here) and [guidelines for contributing](add link here) to this repository.

- [ ] My code follows the code style of this project.
- [ ] This PR does not contain plagiarized content.
- [ ] The title and description of the PR is clear and explains the approach.
- [ ] I am making a pull request against the **main branch** (left side).
- [ ] My commit messages styles matches our requested structure.
- [ ] My code additions will fail neither code linting checks nor unit test.
- [ ] I am only making changes to files I was requested to.

### CODE REVIEW

File: .github/workflows/prod.yml
Comment (incredible-phoenix246) [inline .github/workflows/prod.yml:1]: first of y comment on this

File: src/index.ts
Comment (incredible-phoenix246) [inline src/index.ts:1]: why comment out the notification controller it was working

File: src/middleware/auth.ts
Comment (incredible-phoenix246) [inline src/middleware/auth.ts:1]: this has been declared in the types declaration file why double declare

File: src/routes/notificationsettings.ts
Comment (incredible-phoenix246) [inline src/routes/notificationsettings.ts:1]: why

File: src/utils/logger.ts
Comment (incredible-phoenix246) [inline src/utils/logger.ts:1]: hmmm

**Review by incredible-phoenix246 — CHANGES_REQUESTED** [review]
also we are using yarn what is package-lock doing here

**Review by incredible-phoenix246 — APPROVED** [review]

**Review by Idimmusix — APPROVED** [review]

**Pull request merged**